adminSiteClient/ImagesIndexPage.tsx

```diff
@@ -0,0 +1,260 @@
+                    image={image}
+                    patchImage={api.patchImage}
+                />
+            ),
+        },
+        {
+            title: "Width",
+            dataIndex: "originalWidth",
+            key: "originalWidth",
+            sorter: (a, b) =>
+                a.originalWidth && b.originalWidth
+                    ? a.originalWidth - b.originalWidth
+                    : 0,
+            width: 100,
+        },
+        {
+            title: "Height",
+            dataIndex: "originalHeight",
+            key: "originalHeight",
+            sorter: (a, b) =>
+                a.originalHeight && b.originalHeight
+                    ? a.originalHeight - b.originalHeight
+                    : 0,
+            width: 100,
+        },
+        {
+            title: "Last updated",
+            dataIndex: "updatedAt",
+            key: "updatedAt",
+            width: 150,
+            defaultSortOrder: "descend",
+            sorter: (a, b) =>
+                a.updatedAt && b.updatedAt ? a.updatedAt - b.updatedAt : 0,
+            render: (time) => <Timeago time={time} />,
+        },
+        {
+            title: "Action",
+            key: "action",
+            width: 100,
+            render: (_, image) => (
+                <Space size="middle">
+                    <Button
+                        type="text"
+                        danger
+                        onClick={() => api.deleteImage(image)}
+                    >
+                        Delete
+                    </Button>
+                </Space>
+            ),
+        },
+    ]
+}
+
+function ImageUploadButton({
+    setImages,
+    admin,
+}: {
+    setImages: React.Dispatch<React.SetStateAction<DbEnrichedImage[]>>
+    admin: Admin
+}) {
+    function uploadImage({ file }: { file: string | Blob | RcFile }) {
+        if (typeof file === "string") return
+
+        const reader = new FileReader()
+        reader.onload = async () => {
+            const base64Data = reader.result?.toString()
+
+            const payload = {
+                filename: file.name,
+                content: base64Data,
+                type: file.type,
+            }
+
+            const { image } = await admin.requestJSON<{
+                sucess: true
+                image: DbEnrichedImage
+            }>("/api/image", payload, "POST")
+
+            setImages((images) => [image, ...images])
+        }
+        reader.readAsDataURL(file)
+    }
+    return (
+        <Upload showUploadList={false} customRequest={uploadImage}>
+            <Button type="primary" icon={<FontAwesomeIcon icon={faUpload} />}>
+                Upload image
+            </Button>
+        </Upload>
+    )
+}
+
+export function ImageIndexPage() {
+    const { admin } = useContext(AdminAppContext)
+    const [images, setImages] = useState<DbEnrichedImage[]>([])
+    const [filenameSearchValue, setFilenameSearchValue] = useState("")
+    const api = useMemo(
```

Comment:
This is an unusual usage of `useMemo`. Is there a reason why we do this? It doesn't seem like we really need it, and if we do, `useCallback` would be IMO nicer.